### PR TITLE
[JENKINS-67063] Revert canonicalization of entries when untarring

### PR DESCRIPTION
**Pull request by daniel-beck**
See [JENKINS-67063](https://issues.jenkins-ci.org/browse/JENKINS-67063).

Alternative to #5886. Basically reverts all of the canonicalization added for validation in 63cde2daadc705edf086f2213b48c8c547f98358. As there's a canonical path check further down, this one should not be necessary.

### Proposed changelog entries

* JENKINS-67063: Do not attempt to canonicalize tar entries when untaring, as the result may be unexpected for symlinks.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by jglick** [comment]
> there's a canonical path check further down

Source link?

File: core/src/main/java/hudson/FilePath.java
Comment (jglick) [inline core/src/main/java/hudson/FilePath.java:2834]:
```suggestion
```

File: core/src/main/java/hudson/FilePath.java
Comment (jglick) [inline core/src/main/java/hudson/FilePath.java:2835]: Looks like old litter too.

```suggestion
```

**Review by jglick — APPROVED** [review]
If you say this was not necessary for security, then fine. I suppose existing test coverage would prove it?

File: core/src/main/java/hudson/FilePath.java
Comment (daniel-beck) [reply core/src/main/java/hudson/FilePath.java:2834]: Code as is minimizes diff to before 63cde2d.

**Comment by daniel-beck** [comment]
> Source link?

`#writing` etc. go through https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/SoloFilePathFilter.java, whose `base` is a https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/FilePathFilterAggregator.java, which contains a https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/security/s2m/AdminFilePathFilter.java (which holds bundled default and admin-customized rules) which calls into https://github.com/jenkinsci/jenkins/blob/bea3cef24f885d9a15a387eef1af8c6e46ed6364/core/src/main/java/jenkins/security/s2m/AdminWhitelistRule.java which ends up at https://github.com/jenkinsci/jenkins/blob/bea3cef24f885d9a15a387eef1af8c6e46ed6364/core/src/main/java/jenkins/security/s2m/FilePathRuleConfig.java#L97 to ensure the canonical path to the file being accessed is checked.

`RunningBuildFilePathFilter` separately canonicalizes in https://github.com/jenkinsci/jenkins/blob/bea3cef24f885d9a15a387eef1af8c6e46ed6364/core/src/main/java/jenkins/security/s2m/RunningBuildFilePathFilter.java#L95

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
